### DISCUSSION TITLE
Unable to install ruby 3.3.0 on brand new Fedora 38 VM

### DISCUSSION THREAD

**Discussion by robacarp** [rbenv/ruby-build#2463] [Build failures] [answered]
On a [new fedora vm](https://mac.getutm.app/gallery/fedora-38) I installed asdf with git, and setup ruby with `asdf plugin-add ruby`. Then `asdf install ruby 3.3.0` -- but it failed. The log output was enough information to guide me on my way, but my experience (on macos) has always been that it Just Worked with asdf.

In the interest of helping others along, this is a list of error messages and what I had to do to get it working: 

> Can't locate FindBin.pm in @INC (you may need to install the FindBin module) 
- `yum install perl` (seems a bit much, but that's openssl's thing not ruby!)

> fatal error: zlib.h: No such file or directory
- `yum install zlib-devel`

> fiddle:
>	Could not be configured. It will not be installed.
> [...] missing libffi. Please install libffi or use --with-libffi-source-dir with libffi source location.
- `yum install libffi-devel`

At this point my VM ran out of disk in /tmp because each build I was attempting was downloading a bunch of stuff from scratch, which seems egregious, but I digress. 

> psych:
>	Could not be configured. It will not be installed.
>	Check /tmp/ruby-build.[blah]ruby-3.3.0/ext/psych/mkmf.log for more details.

(in that log file)
> conftest.c:3:10: fatal error: yaml.h: No such file or directory
>    3 | #include <yaml.h>
- `yum install libyaml-devel`

At which point I was able to get ruby 3.3.0 to install! Hopefully this journey is helpful to the project, or to someone else wandering around blindly trying to get ruby running on Fedora.

Hopefully this is help

**Comment by mislav** [comment, marked as answer]
> At which point I was able to get ruby 3.3.0 to install! Hopefully this journey is helpful to the project, or to someone else wandering around blindly trying to get ruby running on Fedora.

Thanks for sharing what worked for you! Note that our wiki also collects community-maintained information about which packages might be prerequisites for building Ruby: https://github.com/rbenv/ruby-build/wiki

> At this point my VM ran out of disk in /tmp because each build I was attempting was downloading a bunch of stuff from scratch, which seems egregious, but I digress.

For future reference, from [our README](https://github.com/rbenv/ruby-build?tab=readme-ov-file#custom-build-configuration):

> RUBY_BUILD_CACHE_PATH: Where to cache downloaded package files. (Default: `~/.rbenv/cache` if invoked as rbenv plugin)
